### ISSUE TITLE
keypoints lost while using moasic  and  RandomPerspective  together

### ISSUE THREAD

**Issue by Hua2544** [ultralytics/ultralytics#20811]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Augmentation

### Bug

 keypoints lost while using moasic  and  RandomPerspective  together  

![Image](https://github.com/user-attachments/assets/bc9687c9-0240-45cd-beda-8537ca7b4ef6)，

![Image](https://github.com/user-attachments/assets/2f450d71-9052-4dab-8bea-c5f822097198)，The possible reason I have located is due to changes in borders or imgsz.

### Environment

Ultralytics 8.3.118 🚀 Python-3.13.2 torch-2.7.0+cu118 CUDA:0 (NVIDIA GeForce RTX 4090, 24210MiB)  task=pose, mode=train, model=/root/mypronoise/ultralytics/cfg/models/11/yolo11-pose.yaml, data=/root/shared-nvme/datasets/noise/pose/stratified_output/mytrain.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=yolo11-pose-202505260927, exist_ok=False, pretrained=False, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, cfg=None, tracker=botsort.yaml, 

### Minimal Reproducible Example

from ultralytics.models.yolo import YOLO
from datetime import datetime

# 获取当前时间
now = datetime.now()
# 格式化为 YYYYMMDDHHMM
time_str = now.strftime("%Y%m%d%H%M")
# print(time_str)

if __name__ == '__main__':
    # model=YOLO(model="ultralytics/cfg/models/v8/myyolov8-GSConv.yaml")
    # modelname="/root/runs/detect/yolo11-202504280445/weights/last.pt"
    modelname="/root/mypronoise/ultralytics/cfg/models/11/yolo11-pose.yaml"
    subname=modelname.split("/")[-1].split(".")[0] +"-"+time_str
    # print(subname)
    model = YOLO(modelname)
    # model.load('yolov8n.pt')
    result = model.train(data=r'/root/shared-nvme/datasets/noise/pose/stratified_output/mytrain.yaml', #每次记得清理缓存防止数据集不对
                        epochs=100, 
                        batch=16,
                        imgsz=640,
                        # optimizer="AdamW",
                        # lr0=0.01 ,# (float) initial learning rate (i.e. SGD=1E-2, Adam=1E-3)
                        # momentum= 0.937 ,
                        # SGD(lr=0.01, momentum=0.9)
                        # optimizer="AdamW",
                        # lr0=0.01 ,# (float) initial learning rate (i.e. SGD=1E-2, Adam=1E-3)
                        # momentum= 0.937 ,
                        # hsv_h=0.0, # (float) image HSV-Hue augmentation (fraction)
                        # hsv_s=0.0, # (float) image HSV-Saturation augmentation (fraction)
                        # hsv_v=0.0 ,# (float) image HSV-Value augmentation (fraction)
                        # degrees=0.0, # (float) image rotation (+/- deg)jnjhbv  
                        # translate=0.0, # (float) image translation (+/- fraction)
                        # scale=0.5, # (float) image scale (+/- gain)
                        # shear=0.0 ,# (float) image shear (+/- deg)
                        # perspective=0.0 ,# (float) image perspective (+/- fraction), range 0-0.001
                        # flipud=0.0, # (float) image flip up-down (probability)
                        # fliplr=0.0, # (float) image flip left-right (probability)
                        # bgr=0.0 ,# (float) image channel BGR (probability)
                        # mosaic=0.0 ,# (float) image mosaic (probability)
                        # rect=True,
                        # mixup=0.0, # (float) image mixup (probability)
                        # copy_paste=0.0, # (float) segment copy-paste (probability)
                        # copy_paste_mode="flip" ,# (str) the method to do copy_paste augmentation (flip, mixup)
                        # auto_augment="randaugment", # (str) auto augmentation policy for classification (randaugment, autoaugment, augmix)
                        # erasing=0.0 ,# (float) probability of random erasing during classification training (0-0.9), 0 means no erasing, must be less than 1.0.
                        device=0,
                        pretrained=False,
                        name=subname,
                        # resume=True
                        )

### Additional

I have confirmed that there are no errors in the label and it is displayed normally.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
Can you provide a minimum reproducible example?

**Comment by Y-T-G** [comment]
Is this issue with original Ultralytics code or did you modify the code?

**Comment by Hua2544** [comment]
i use AnimalPose public dataset in the net， and i fix the incorrect keypoint in it. you can use that.
